### DISCUSSION TITLE
Rxjs compatibility with old browsers

### DISCUSSION THREAD

**Discussion by dovranJorayev** [ReactiveX/rxjs#7404] [Q&A] [answered]
Help me please to define is rxjs latest versions usable with old browsers (e.g chromium v65-75), or i forced to use old versions of rxjs (e.g v4-v5). Tried to search on internet and local communities but could not get an answer?

**Comment by benlesh** [comment]
I guess I don't know what you're trying to do that isn't compatible?

**Reply by dovranJorayev** [reply]
That mean i can use all of the operators from rxjs latest version, and it will just work in old environment (e.g chromium v65-75 while actual is v120). 
I plan to build a library with using rxjs as it core. And i have a requirement to build it as 2 targets. 
- mylib
- mylib/compat
I am asking because i am afraid i need to use for that purpose not same version of rxjs

**Reply by dovranJorayev** [reply]
I will need to add polyfills for promises, transpile code  to old syntax but it is build time overhead. Maybe i am missing something...

**Reply by benlesh** [reply, marked as answer]
Whoa... polyfills for promises? So you're talking like 2013 or earlier? You might want to use RxJS 5 if that's the case, because I think in v6 or v7 we stopped supporting browsers that don't have promise implementations. Then again, if you polyfill that, I'd expect things to work with RxJS 7 and 8.

**Reply by dovranJorayev** [reply]
Not like old browsers, but Smart TV web apps. Smart TVs even new ones suits with old browser engines (e.g Chromium 70-75, when current 120+). Technically you are right, it is old browsers.
So i guess answer is 
`In theory, if all new features of JS will be polifiled in that environment, i can use even latest versions of rxjs`.
So, correct answer will found in practice only :)
